src/character.cpp

```diff
@@ -1512,6 +1513,28 @@ bool Character::is_wearing_active_power_armor() const
     return false;
 }
 
+void layer_details::reset() {
```

Comment:
Seems a simple `*this = layer_details();` would do the same. You can do the same with `encumbrance_data::reset`.